tools/serverpod_cli/lib/src/analyzer/entities/entity_dependency_resolver.dart

```diff
@@ -291,19 +291,37 @@ class EntityDependencyResolver {
     } else {
       var foreignFields = referenceClass.fields.where((field) {
         var fieldRelation = field.relation;
-        if (fieldRelation is! ForeignRelationDefinition) return false;
-        return fieldRelation.parentTable == classDefinition.tableName &&
-            fieldRelation.name == relation.name;
+        if (!(fieldRelation is UnresolvedObjectRelationDefinition ||
+            fieldRelation is ForeignRelationDefinition)) return false;
```

Comment:
Suggestion: Place negation as close to check as possible.

Always hard to parse the impact on a negation on a group of operations. I think it is a bit simpler to be able to reach each check individually. (Extra line break after to separate the two returns.)

```suggestion
        if (fieldRelation is! UnresolvedObjectRelationDefinition &&
            fieldRelation is! ForeignRelationDefinition) {
             return false;
        }
        
```